src/Streamly/Coreutils/RealPath.hs

```diff
@@ -0,0 +1,21 @@
+-- |
+-- Module      : Streamly.Coreutils.RealPath
+-- Copyright   : (c) 2022 Composewell Technologies
+-- License     : BSD-3-Clause
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+-- Returns resolved symbolic link target.
+
+module Streamly.Coreutils.RealPath
+    (realPath)
+where
+
+import System.Directory (canonicalizePath )
+
+-- | Make a path absolute, normalize the path,
+-- and remove as many indirections from it as possible.
+
+realPath :: FilePath -> IO FilePath
+realPath = canonicalizePath
```

Comment:
This is not the same. Did you test and compare the realpath coreutil and this?

Reply:
yes:
*Streamly.Coreutils.RealPath> realPath "./test2.txt"
"/home/ranjeetr/composewell/CoreUtils/streamly-coreutils/test2.txt"
*Streamly.Coreutils.RealPath> realPath "../test2.txt"
"/home/ranjeetr/composewell/CoreUtils/test2.txt"
*Streamly.Coreutils.RealPath> 

kaveri:~/composewell/CoreUtils/streamly-coreutils  (monadic-predicates)$ (nix) realpath ./test2.txt
/home/ranjeetr/composewell/CoreUtils/streamly-coreutils/test2.txt
kaveri:~/composewell/CoreUtils/streamly-coreutils  (monadic-predicates)$ (nix) realpath ../test2.txt
/home/ranjeetr/composewell/CoreUtils/test2.txt
kaveri:~/composewell/CoreUtils/streamly-coreutils  (RealPath_module)$ (nix)

Reply:
same behaviors for symlink and nonexistent files.
*Streamly.Coreutils.RealPath> realPath "./ls.txt"
"/home/ranjeetr/composewell/CoreUtils/streamly-coreutils/test2.txt"
*Streamly.Coreutils.RealPath> realPath "../lsssss.txt"
"/home/ranjeetr/composewell/CoreUtils/lsssss.txt